### PR TITLE
Release notes for Foreman 2.4.0

### PR DESCRIPTION
**Pull request by upadhyeammit**

### CODE REVIEW

File: _includes/manuals/2.4/1.2_release_notes.md [hunk 1]
```diff
@@ -192,30 +214,30 @@ See the upgrade warnings for more information.
...
+* Foreman content proxy doesn't listen on port 80 anymore ([#31662](https://projects.theforeman.org/issues/31662))
 * The container gateway smart proxy plugin needs installer support ([#31642](https://projects.theforeman.org/issues/31642))
+* Deploy Pulp 3 by default on Foreman server and as a mirror configuration on content proxies ([#31614](https://projects.theforeman.org/issues/31614))
+* foreman-proxy-certs-generate fails with custom certificates ([#31611](https://projects.theforeman.org/issues/31611))
 * Increase Artemis disk usage limit ([#31607](https://projects.theforeman.org/issues/31607))
 * Allow changing some Foreman Proxy and Apache hostnames on Katello-based certificates ([#31509](https://projects.theforeman.org/issues/31509))
 * Update candlepin broker.xml for jobs processing ([#31497](https://projects.theforeman.org/issues/31497))
 * Remove digital ocean plugin ([#31490](https://projects.theforeman.org/issues/31490))
+* Upgrade results in empty /var/opt/rh/rh-postgresql12/lib/pgsql/data which causes upgrade to fail ([#31486](https://projects.theforeman.org/issues/31486))
```
Comment (ekohl) [inline _includes/manuals/2.4/1.2_release_notes.md]: I changed this category

File: _includes/manuals/2.4/1.2_release_notes.md [hunk 2]
```diff
@@ -192,30 +214,30 @@ See the upgrade warnings for more information.
...
 * Create qdrouter inbound link for katello.agent queue ([#31800](https://projects.theforeman.org/issues/31800))
 * Create pulp.task queue for katello agent ([#31761](https://projects.theforeman.org/issues/31761))
 * Remove foreman-proxy-plugin-pulp from Foreman scenario ([#31757](https://projects.theforeman.org/issues/31757))
 * Restore katello::qpid_client ([#31693](https://projects.theforeman.org/issues/31693))
+* Command exceeded timeout while Installer executes foreman-rake db:migrate ([#31670](https://projects.theforeman.org/issues/31670))
+* Foreman content proxy doesn't listen on port 80 anymore ([#31662](https://projects.theforeman.org/issues/31662))
 * The container gateway smart proxy plugin needs installer support ([#31642](https://projects.theforeman.org/issues/31642))
+* Deploy Pulp 3 by default on Foreman server and as a mirror configuration on content proxies ([#31614](https://projects.theforeman.org/issues/31614))
+* foreman-proxy-certs-generate fails with custom certificates ([#31611](https://projects.theforeman.org/issues/31611))
```
Comment (ekohl) [inline _includes/manuals/2.4/1.2_release_notes.md]: I changed this category

File: _includes/manuals/2.4/1.2_release_notes.md [hunk 3]
```diff
@@ -192,30 +214,30 @@ See the upgrade warnings for more information.
...
 * Increase Artemis disk usage limit ([#31607](https://projects.theforeman.org/issues/31607))
 * Allow changing some Foreman Proxy and Apache hostnames on Katello-based certificates ([#31509](https://projects.theforeman.org/issues/31509))
 * Update candlepin broker.xml for jobs processing ([#31497](https://projects.theforeman.org/issues/31497))
 * Remove digital ocean plugin ([#31490](https://projects.theforeman.org/issues/31490))
+* Upgrade results in empty /var/opt/rh/rh-postgresql12/lib/pgsql/data which causes upgrade to fail ([#31486](https://projects.theforeman.org/issues/31486))
+* Ensure import/export directories are created and have proper permissions ([#31468](https://projects.theforeman.org/issues/31468))
 * Drop ssl_protocol parameter on foreman_proxy_content ([#31435](https://projects.theforeman.org/issues/31435))
 * Installer can't install the Ansible plugin on Debian/Ubuntu ([#31430](https://projects.theforeman.org/issues/31430))
+* Unable to set ping_free_ip in Foreman Proxy's dhcp.yml ([#31415](https://projects.theforeman.org/issues/31415))
 * Disable TLS 1.0 and 1.1 by default in Apache ([#31387](https://projects.theforeman.org/issues/31387))
```
Comment (ekohl) [inline _includes/manuals/2.4/1.2_release_notes.md]: I changed this category

File: _includes/manuals/2.4/1.2_release_notes.md [hunk 4]
```diff
@@ -103,18 +112,24 @@ See the upgrade warnings for more information.
...
 #### Foreman - Puppet Reports
 * WebUI spinner spins forever when clicking a column within config report ([#31526](https://projects.theforeman.org/issues/31526))
 
 #### Foreman - Puppet integration
 * The fix for #31168 overwrites $::hostname with the FQDN in Puppet ([#31882](https://projects.theforeman.org/issues/31882))
 * Puppet Environments are not listed properly in the SCP details page ([#31625](https://projects.theforeman.org/issues/31625))
+* Improve /puppetrun error message ([#31543](https://projects.theforeman.org/issues/31543))
 * Each smart class parameter appears twice, when having two environments ([#31538](https://projects.theforeman.org/issues/31538))
 * TemplateInput type registry ([#31284](https://projects.theforeman.org/issues/31284))
+* Add warning banner about puppet ENC extraction ([#30215](https://projects.theforeman.org/issues/30215))
```
Comment (ekohl) [inline _includes/manuals/2.4/1.2_release_notes.md:131]: I think this deserves a section in deprecations.

**Review by ekohl — COMMENTED** [review]
I think I dislike the deprecation warning. I opened https://community.theforeman.org/t/how-to-deal-with-user-hostile-warnings/22831.

File: _includes/manuals/2.4/1.2_release_notes.md [hunk 5]
```diff
@@ -10,42 +14,28 @@ Additionally, it is stricter in reverse proxy handling. Out of the box Rails acc
...
+####  Personal Access Tokens
+
+WebUI can be used to handle Personal Access Tokens. A new token can be generated by the user and set to expire at a certain time and date.
+To make tokens void before their expiration date and time, one can use the revoke action. ([31080](https://projects.theforeman.org/issues/31080))
+
+* New tab: Hover to current logged in user -> _My Account_ -> _Personal Access Tokens_
+
+#### Support for Redfish to BMC Smart Proxy
 
+The Intelligent Platform Management Interface (IPMI) is being phased out in favour of Redfish for network hardware management. Some of the features for example, boot media aren't available across all vendors (HPE and Dell in particular). The Redfish provider for the BMC Smart Proxy solves these issues while also potentially allowing additional BMC functions to be integrated. ([29636](https://projects.theforeman.org/issues/29636))
```
Comment (tbrisker) [inline _includes/manuals/2.4/1.2_release_notes.md]: @lzap can you take a look at this one please?

File: _includes/manuals/2.4/1.2_release_notes.md [hunk 6]
```diff
@@ -10,42 +14,28 @@ Additionally, it is stricter in reverse proxy handling. Out of the box Rails acc
...
-* Show host name in ENC output  ([#31168](https://projects.theforeman.org/issues/31168))
-* Use single quotes in registration templates for curl ([#31143](https://projects.theforeman.org/issues/31143))
-* Developer Docs - Create plugin - host.refresh_status & subscriber ([#31139](https://projects.theforeman.org/issues/31139))
-* Add global css registration from plugins ([#30978](https://projects.theforeman.org/issues/30978))
-* Allow Ubuntu 18.04 to netboot with a VLAN tag ([#30737](https://projects.theforeman.org/issues/30737))
-* Add warning banner about puppet ENC extraction ([#30215](https://projects.theforeman.org/issues/30215))
-* Add support for Redfish to BMC Smart Proxy ([#29636](https://projects.theforeman.org/issues/29636))
+####  Personal Access Tokens
+
+WebUI can be used to handle Personal Access Tokens. A new token can be generated by the user and set to expire at a certain time and date.
```
Comment (ekohl) [inline _includes/manuals/2.4/1.2_release_notes.md]: Maybe it's good to clarify that previously it only had an API.

File: _includes/manuals/2.4/1.2_release_notes.md (hunk 5 above)
Comment (ekohl) [reply _includes/manuals/2.4/1.2_release_notes.md]: Are there plans to phase out IPMI? Not all hardware supports Redfish so I think phasing out is too strong.

Comment (lzap) [reply _includes/manuals/2.4/1.2_release_notes.md]: No there are no plans at the moment, Redfish was just contributed by one of our community users and it is very basic. We currently have not plans on giving up IPMI.

Basically just say that Redfish is now available as another BMC provider.

File: _includes/latest_news.html [hunk 7]
```diff
@@ -1,2 +1,3 @@
+<p>Foreman 2.4.0 has been released! <a href="manuals/2.4/quickstart_guide.html"> Follow the quick start to install it.</a></p>
 <p>Foreman 2.3.3 has been released! <a href="manuals/2.3/quickstart_guide.html"> Follow the quick start to install it.</a></p>
 <p>Foreman 2.2.3 has been released! <a href="manuals/2.2/quickstart_guide.html"> Follow the quick start to install it.</a></p>
```
Comment (ekohl) [inline _includes/latest_news.html]: Should 2.2 be removed now?

File: _includes/manuals/2.4/1.2_release_notes.md [hunk 8]
```diff
@@ -2,6 +2,10 @@
 
 ### Headline features
 
+#### Ansible inventory template
+
+We've already eliminated the necessity for Katello to use Ansible Inventory[31193](https://projects.theforeman.org/issues/31193). This change eliminates the need for the Ansible plugin to render Ansible inventory. ([31482](https://projects.theforeman.org/issues/31482))
```
Comment (ekohl) [inline _includes/manuals/2.4/1.2_release_notes.md]:
```suggestion
We've already eliminated the necessity for Katello to use Ansible Inventory ([#31193](https://projects.theforeman.org/issues/31193)). This change eliminates the need for the Ansible plugin to render Ansible inventory. ([#31482](https://projects.theforeman.org/issues/31482))
```

File: _includes/manuals/2.4/1.2_release_notes.md [hunk 9]
```diff
@@ -10,42 +14,27 @@ Additionally, it is stricter in reverse proxy handling. Out of the box Rails acc
...
-* Show host name in ENC output  ([#31168](https://projects.theforeman.org/issues/31168))
-* Use single quotes in registration templates for curl ([#31143](https://projects.theforeman.org/issues/31143))
-* Developer Docs - Create plugin - host.refresh_status & subscriber ([#31139](https://projects.theforeman.org/issues/31139))
-* Add global css registration from plugins ([#30978](https://projects.theforeman.org/issues/30978))
-* Allow Ubuntu 18.04 to netboot with a VLAN tag ([#30737](https://projects.theforeman.org/issues/30737))
-* Add warning banner about puppet ENC extraction ([#30215](https://projects.theforeman.org/issues/30215))
-* Add support for Redfish to BMC Smart Proxy ([#29636](https://projects.theforeman.org/issues/29636))
+####  Personal Access Tokens
+
+The API endpoints were introduced for Personal Access Tokens in Foreman 1.17 version. Now WebUI can be used to manage Personal Access Tokens. A new token can be generated by the user and set to expire at a certain time and date. To make tokens void before their expiration date and time, one can use the revoke action. ([31080](https://projects.theforeman.org/issues/31080))
```
Comment (ekohl) [inline _includes/manuals/2.4/1.2_release_notes.md]:
```suggestion
Personal Access Tokens were introduced in Foreman 1.17 version with API endpoints only. Now WebUI can be used to manage Personal Access Tokens. A new token can be generated by the user and set to expire at a certain time and date. To make tokens void before their expiration date and time, one can use the revoke action. ([#31080](https://projects.theforeman.org/issues/31080))
```

File: _includes/manuals/2.4/1.2_release_notes.md [hunk 10]
```diff
@@ -10,42 +14,27 @@ Additionally, it is stricter in reverse proxy handling. Out of the box Rails acc
...
+
+* New tab: Hover to current logged in user -> _My Account_ -> _Personal Access Tokens_
+
+#### Support for Redfish to BMC Smart Proxy
+
+The one more BMC provider Redfish is now available. Some of the features for example, boot media aren't available across all vendors (HPE and Dell in particular). The Redfish provider for the BMC Smart Proxy solves this issue while also potentially allowing additional BMC functions to be integrated. ([29636](https://projects.theforeman.org/issues/29636))
```
Comment (ekohl) [inline _includes/manuals/2.4/1.2_release_notes.md]:
```suggestion
Redish is now available as another BMC provider. Some of the features, such as boot media, weren't available across all vendors (HPE and Dell in particular). The Redfish provider for the BMC Smart Proxy solves this issue while also potentially allowing additional BMC functions to be integrated. ([#29636](https://projects.theforeman.org/issues/29636))
```

File: _includes/version.html [hunk 11]
```diff
@@ -1,10 +1,10 @@
 <div class="version-marker">
   <p>
     Latest versions:
+      <a href="{{ site.baseurl }}manuals/2.4/index.html#Releasenotesfor2.4">2.4.0</a>
+      /
       <a href="{{ site.baseurl }}manuals/2.3/index.html#Releasenotesfor2.3">2.3.3</a>
       /
       <a href="{{ site.baseurl }}manuals/2.2/index.html#Releasenotesfor2.2">2.2.3</a>
```
Comment (ekohl) [inline _includes/version.html]: Looks like I missed this one, but can also be removed.

**Review by ekohl — APPROVED** [review]
Perhaps @melcorr can give it a read through for grammar and spelling, but looks good to me.

File: _includes/manuals/2.4/1.2_release_notes.md [hunk 12]
```diff
@@ -10,42 +14,27 @@ Additionally, it is stricter in reverse proxy handling. Out of the box Rails acc
...
-* Show host name in ENC output  ([#31168](https://projects.theforeman.org/issues/31168))
-* Use single quotes in registration templates for curl ([#31143](https://projects.theforeman.org/issues/31143))
-* Developer Docs - Create plugin - host.refresh_status & subscriber ([#31139](https://projects.theforeman.org/issues/31139))
-* Add global css registration from plugins ([#30978](https://projects.theforeman.org/issues/30978))
-* Allow Ubuntu 18.04 to netboot with a VLAN tag ([#30737](https://projects.theforeman.org/issues/30737))
-* Add warning banner about puppet ENC extraction ([#30215](https://projects.theforeman.org/issues/30215))
-* Add support for Redfish to BMC Smart Proxy ([#29636](https://projects.theforeman.org/issues/29636))
+####  Personal Access Tokens
+
+Personal Access Tokens were introduced in Foreman 1.17 version with API endpoints only. Now WebUI can be used to manage Personal Access Tokens. A new token can be generated by the user and set to expire at a certain time and date. To make tokens void before their expiration date and time, one can use the revoke action. ([#31080](https://projects.theforeman.org/issues/31080))
```
Comment (melcorr) [inline _includes/manuals/2.4/1.2_release_notes.md]:
```suggestion
Personal Access Tokens were introduced in Foreman 1.17 version with API endpoints only. Now, you can manage Personal Access Tokens through the Foreman web UI. You can generate a new token and set it to expire at a certain date and time. To make tokens void before their expiration date and time, you can use the revoke action. ([#31080](https://projects.theforeman.org/issues/31080))
```

**Review by melcorr — COMMENTED** [review]
One suggested change, but other than that it LGTM!

**Pull request merged**